components/src/dynamo/vllm/args.py

```diff
@@ -154,9 +188,19 @@ def parse_args() -> Config:
         # This becomes an `Option` on the Rust side
         config.served_model_name = None
 
-    config.namespace = os.environ.get("DYN_NAMESPACE", "dynamo")
-    config.component = "prefill" if args.is_prefill_worker else "backend"
-    config.endpoint = "generate"
+    if args.endpoint:
+        # User provided explicit endpoint
+        config.namespace, config.component, config.endpoint = parse_endpoint(
+            args.endpoint
+        )
+    else:
+        # Default: use model name in endpoint for uniqueness
+        # This ensures different models register as different services
```

Comment:
Do we have a latent bug waiting to be hit in sglang/trtllm too for the scenario you ran into?